### ISSUE TITLE
Expose navigation button event handlers as external API methods.

### ISSUE THREAD

**Issue by aeschylus** [openseadragon/openseadragon#489]
One benefit of this is enabling the use of the functions in externally-specified custom controls.

**Description:**
There are highly flexible functions exposed that accept specific parameters for zooming, panning, and navigating (zoomBy, panBy, zoomTo, fitBounds, etc.), but it would be useful to API users to directly call the functions used by the OSD control buttons directly. For instance, the "zoom in" button doesn't simply zoom in by a certain amount. It takes into account the total extent of possible zooming, as well as the canvas constraints and current position. This more rigid functionality would be useful to be able to call directly, for triggering navigation from outside the standard controls.

**Relevant source:** https://github.com/openseadragon/openseadragon/blob/master/src/viewer.js#L2848

**Comment by iangilman** [comment]
Updating code link:

https://github.com/openseadragon/openseadragon/blob/2ed99077eb364884103b61285975017071fa210a/src/viewer.js#L3390

**Comment by achu1998** [comment]
Hi @iangilman , If the issue is still open, I'd like to take a look. Please point me in the right direction.

**Comment by iangilman** [comment]
Looks like above I proposed some possible names: zoomIn(), startZoom(), and endZoom().

**Comment by achu1998** [comment]
Yes, I got it. Do we need to rename it? Since the functions are being called in the bindStandardControls handlers or You suggest renaming both 

endZoomingHandler       = $.delegate( this, endZooming ),
doSingleZoomInHandler   = $.delegate( this, doSingleZoomIn ),
beginZoomingOutHandler  = $.delegate( this, beginZoomingOut ),
doSingleZoomOutHandler  = $.delegate( this, doSingleZoomOut ),

**Comment by achu1998** [comment]
@iangilman I have created a pull request [here](https://github.com/openseadragon/openseadragon/pull/2702). Please take a look and let me know if any changes need to be done.
